src/tdyregression.c

```diff
@@ -22,12 +22,11 @@ PetscErrorCode TDyRegressionInitialize(TDy tdy) {
   PetscFunctionBegin;
   TDY_START_FUNCTION_TIMER()
 
-  dm = tdy->dm;
 
   ndof_per_cell = 1;
   if (tdy->mode == TH) ndof_per_cell = 2;
 
-  regression = (TDy_regression *) malloc(sizeof(TDy_regression));
+  regression = (TDyRegression *) malloc(sizeof(TDyRegression));
```

Comment:
Do we want to use `malloc` here, or the PETSc equivalent? I guess we haven't discussed this much. In any case, you don't need to cast the results of `malloc` to the type of pointer you're using--that's a C++ thing. Of course, if this is what's done in PETSc, then that's fine.

Reply:
This comment applies to all similar cases above. :-)

Reply:
How about changing `malloc` to `PetscMalloc` at multiple locations in a new PR?